### PR TITLE
123 ssi android build to s3

### PR DESCRIPTION
**Pull request by sKudryashov**
This PR creates a webhook in GitHub actions that allows building FFI for android and pushes built files to AWS S3 the same way it is implemented in core. The difference is that this pipeline doesn't use cache due to the relatively low frequency of usage SSI FFI builds.
To test FFI build yourself you can go here: https://github.com/knox-networks/ssi/actions/workflows/release_ffi.yaml and on the top right corner run workflow "build FFI" on a current branch: 123-ssi-android-build-to-s3.

### CODE REVIEW

File: registry_resolver/build.rs [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
-    let package_path = "buf.build/knox-networks/registry-mgmt";
-    let commit = "f7ff6f57030c418e886459a18b35645e";
-    let std::process::Output { status, stderr, .. } = std::process::Command::new("buf")
-        .arg("generate")
-        .arg(format!("{package_path}:{commit}"))
-        .output()?;
-
-    if !status.success() {
-        return Err(String::from_utf8_lossy(&stderr).into());
+    #[cfg(not(feature = "ci_ffi_build"))]
```
Comment (LuisOsta) [inline registry_resolver/build.rs:2]: Why is this disabled during the ffi build?

**Review by LuisOsta — APPROVED** [review]

File: registry_resolver/build.rs (hunk 1 above)
Comment (sKudryashov) [reply registry_resolver/build.rs:2]: For some strange reason cross is not able to build the project with buf being built in the build.rs. I've tried to figure out why, most likely it lacks some envvars, but eventually workaround with building buf separately, before the FFI build worked out.

Comment (LuisOsta) [reply registry_resolver/build.rs:2]: That's strange, well good to know!

File: .github/workflows/release_ffi.yaml [hunk 2]
```diff
@@ -58,43 +59,35 @@ jobs:
...
+          sudo chown -R "$USER:$USER" "/home/$USER"
       - name: Install deps
         run: ./container/deps.sh --force grpc_health_probe flatc cargo-make protoc buf
       - run: |
           sudo chown -R "$USER:$USER" "/home/$USER"
-      - name: Generate FFI Header
-        run: |
-          source .env.cachepot.local
+
+      - run: which buf
```
Comment (mkatychev) [inline .github/workflows/release_ffi.yaml:80]: should this line be removed?

File: .github/workflows/release_ffi.yaml [hunk 3]
```diff
@@ -58,43 +59,35 @@ jobs:
...
+        
+      - run: cd ./registry_resolver && buf generate buf.build/knox-networks/registry-mgmt:f7ff6f57030c418e886459a18b35645e
+      - run: chmod -R 0777 ./registry_resolver/src/gen
+      
+      - run: |
+          cargo make ffi-build
```
Comment (mkatychev) [inline .github/workflows/release_ffi.yaml:86]:
```suggestion
      - run: cargo make ffi-build
```
A lot of these don't need a `|`

**Review by mkatychev — APPROVED** [review]

**Pull request merged**